core/consensus/src/engine.rs

```diff
@@ -78,13 +78,18 @@ impl<Adapter: ConsensusAdapter + 'static> Engine<FixedPill> for ConsensusEngine<
         let (ordered_tx_hashes, propose_hashes) = self
             .adapter
             .get_txs_from_mempool(
-                ctx,
+                ctx.clone(),
                 next_height,
                 current_consensus_status.cycles_limit,
                 current_consensus_status.tx_num_limit,
             )
             .await?
             .clap();
+        let signed_txs = self
+            .adapter
+            .get_full_txs(ctx.clone(), ordered_tx_hashes.clone())
+            .await?;
+        let order_signed_transactions_hash = digest_signed_transactions(&signed_txs)?;
```

Comment:
The hash value of the signed transaction is calculated at the time of the proposal.